app/models/host/base.rb

```diff
@@ -246,7 +246,11 @@ def overwrite=(value)
     end
 
     def primary_interface
-      get_interface_by_flag(:primary)
+      # Returns a dummy object if host is unmanaged and without interfaces.
+      # Unmanaged hosts still have the interface requirement due to delegations and certain validations.
+      # A refactoring that allows unmanaged hosts to exist without interfaces would be welcome.
+      return get_interface_by_flag(:primary) if managed || interfaces.detect(&:primary)
+      self.interfaces.build(:primary => true, :provision => true, :managed => false, :type => 'Nic::Managed')
```

Comment:
This is pretty much equivalent to the `build_required_interfaces` method called in the constructor. 

....rant.....

Yes.

I also think it couldn't be dirtier but 1.8 is near and we won't have another RC for testing.